aws/resource_aws_service_discovery_http_namespace_test.go

```diff
@@ -0,0 +1,92 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/servicediscovery"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSServiceDiscoveryHttpNamespace_basic(t *testing.T) {
+	resourceName := "aws_service_discovery_http_namespace.test"
+	rName := "terraformtesting"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsServiceDiscoveryHttpNamespaceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccServiceDiscoveryHttpNamespaceConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsServiceDiscoveryHttpNamespaceExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "arn"),
+					resource.TestCheckResourceAttr(resourceName, "name", rName),
+					resource.TestCheckResourceAttr(resourceName, "description", "test"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAwsServiceDiscoveryHttpNamespaceDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).sdconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_service_discovery_http_namespace" {
+			continue
+		}
+
+		input := &servicediscovery.GetNamespaceInput{
+			Id: aws.String(rs.Primary.ID),
+		}
+
+		_, err := conn.GetNamespace(input)
+		if err != nil {
+			if isAWSErr(err, servicediscovery.ErrCodeNamespaceNotFound, "") {
+				return nil
+			}
+			return err
+		}
+	}
+	return nil
+}
+
+func testAccCheckAwsServiceDiscoveryHttpNamespaceExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).sdconn
+
+		input := &servicediscovery.GetNamespaceInput{
+			Id: aws.String(rs.Primary.ID),
+		}
+
+		_, err := conn.GetNamespace(input)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+}
+
+func testAccServiceDiscoveryHttpNamespaceConfig(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_service_discovery_http_namespace" "test" {
+  name = %q
+  description = "test"
```

Comment:
Nit: While its likely handled properly in the resource code already, it'd be nice if `description` was removed in the basic test configuration and tested separately since it is an optional argument.

Reply:
Sorry I probably should have been more clear what I meant. While splitting out the description test was the first step, the next steps were removing this line here in `testAccServiceDiscoveryHttpNamespaceConfig()` so `TestAccAWSServiceDiscoveryHttpNamespace_basic()` doesn't add `Description` to the `CreateHttpNamespace` call, then creating a separate configuration function for usage in the description test:

```go
func testAccServiceDiscoveryHttpNamespaceConfigDescription(rName, description string) string {
	return fmt.Sprintf(`
resource "aws_service_discovery_http_namespace" "test" {
  description = %q
  name        = %q
}
`, description, rName)
}
```

That's my bad. I'll make this adjustment on merge rather than bothering you about it.